assimilation_code/programs/readme.rst

```diff
@@ -0,0 +1,180 @@
+ 
+:doc:`model_mod_check <model_mod_check/model_mod_check>` 
+  Program to test some of the more fundamental routines in any ``model_mod``, especially a for a new model.
+
+:doc:`perturb_single_instance <perturb_single_instance/perturb_single_instance>`
+   Generate an ensemble of perturbed ensemble member restart files.
+   (Alternatively, you might perturb the model state using ``model_nml`` variables).
+ 
+:doc:`gen_sampling_err_table <gen_sampling_err_table/gen_sampling_err_table>`
+   Computes a table of values needed to apply Sampling Error Correction (SEC),
+   which corrects covariances based on small sample size statistics.
+ 
+Creating Observation Sequence Files
+-----------------------------------
+
+obs_utils/create_obs_grid
+   Create a set of observations located on a regular grid.  Obs have no data values.
+
+:doc:`create_fixed_network_seq <create_fixed_network_seq/create_fixed_network_seq>` 
+   Reads observation sequence file information from standard input 
+   and replicates it multiple times in a second observation sequence file, at user specified dates. 
+ 
+obs_utils/obs_timejitter
+   Randomly perturb the times of the observations in a (usually) set_def.out file.
+   Writes the results to (usually) obs_seq.in.
+
+:doc:`create_obs_sequence <create_obs_sequence/create_obs_sequence>`
+   Creates a short andor synthetic observation sequence file using values read from standard input.
+ 
+Querying Observation Sequence Files
+-----------------------------------
+
+obs_utils/obs_info
+   Summarize obs types, times, counts found in observation sequence file(s).
+
+obs_utils/obs_assim_count
+   Prints out a quick table of obs types and counts, overall start and stop times, 
+   and metadata strings and counts.  See obs_diag for more.
+   There is an older version in the obs_assim_count directory.
+
+:doc:`obs_seq_coverage <obs_seq_coverage/obs_seq_coverage>`
+   Queries a set of observation sequence files to determine which observation locations report
+   frequently enough to be useful for a verification study.
+ 
+obs_total_error
+   Prints the total error in the mean and spread from an obs_seq file 
+   which has been through both perfect_model_obs and filter, so it has copies
+   'truth', 'ensemble mean', and 'ensemble spread'.
+   You can get more information by running the obs_diag program.
+
+Changing Observation Sequence Files
+-----------------------------------
+
+:doc:`obs_keep_a_few <obs_keep_a_few/obs_keep_a_few>`
+   Creates an output observation sequence file that is shorter than the input obs_seq file.
+ 
+:doc:`obs_selection <obs_selection/obs_selection>`
+   Extracts observations out of one or more obs_sequence files
+   according to a  list of observation types, times, and locations.
+   The list is usually created by :doc:`obs_seq_coverage/obs_seq_coverage`, 
+   but can be an observation sequence file,
+ 
+:doc:`obs_sequence_tool <obs_sequence_tool/obs_sequence_tool>`
```

Comment:
suggested order for this section:  obs_sequence_tool, obs_common_subset, obs_selection, obs_seq_verify,  obs_loop, obs_keep_a_few, and then the remaining files.